common/app/views/support/cleaner/AmpAdCleaner.scala

```diff
@@ -12,84 +12,80 @@ import scala.xml.Elem
 
 object AmpAdCleaner {
   val AD_LIMIT = 8
-  val CHARS_BETWEEN_ADS = 700
-  val DONT_INTERLEAVE_SMALL_PARA = 50
-
-  def findElementsNeedingAdsAfter(children: List[Element]): List[Element] = {
-
-    val ALLOWED = 0
-    val DISALLOWED = 1
-
-    // create the constraints for each paragraph how far before and after we can have it
-    val constraints = children.map { element =>
-
-      def para = element.tagName() != "p"
-      def short = element.text().length < DONT_INTERLEAVE_SMALL_PARA
-
-      if (para) {
-        // don't put an ad before or too close after an embed
-        AdRepel(300, 200, element)
-      } else if (short) {
-        // don't interleave ads between small paragraphs
-        AdRepel(DISALLOWED, DISALLOWED, element)
-      } else {
-        AdRepel(ALLOWED, ALLOWED, element)
+  val SMALL_PARA_CHARS = 50
+  val MIN_CHAR_BUFFER = 700
+  val IMG_BUFFER_FWD = 300 // really any non-p element type
+  val IMG_BUFFER_BWD = 200
```

Comment:
good but FWD could just be FORWARD etc.  and "buffer" is a bit of an overloaded term, could say GAP or something